modules/extension/xsd/xsd-core/src/main/java/org/geotools/xml/impl/ParseExecutor.java

```diff
@@ -328,11 +334,16 @@ public XSDNamedComponent getDeclaration() {
             // for mixed
             if (instance.getTypeDefinition() instanceof XSDComplexTypeDefinition
                     && ((XSDComplexTypeDefinition) instance.getTypeDefinition()).isMixed()) {
-                //collape the text
-                text = Whitespace.COLLAPSE.preparse(text);
+                //Collapse the text
+                //Seems to be an issue for transaction updates - See GEOS-8227
```

Comment:
I'd say "Collapse the text, but don't do so for CDATA where it's meant to be preserved"
Reference to GeoServer at this point is probably not required